src/bot/setcommands.ts

```diff
@@ -0,0 +1,141 @@
+import type { BotCommand } from "@grammyjs/types";
+import type { CommandContext } from "grammy";
+import { GrammyContext } from "./helpers/grammy-context";
+
+export const BOT_COMMANDS = [
+  {
+    command: "start",
+    description: "View the welcome message",
+    type: "all_private_chats",
+  },
+  {
+    command: "register",
+    description: "Pair your GitHub account",
+    type: "all_private_chats",
+  },
+  {
+    command: "subscribe",
+    description: "Subscribe to notifications",
+    type: "all_private_chats",
+  },
+  {
+    command: "unsubscribe",
+    description: "Unsubscribe from selected notifications",
+    type: "all_private_chats",
+  },
+  {
+    command: "wallet",
+    description: "Register your wallet address",
+    type: "all_private_chats",
+  },
+  {
+    command: "myid",
+    description: "Get your user ID",
+    type: "all_private_chats",
+  },
+  {
+    command: "botid",
+    description: "Get the bot's ID",
+    type: "all_private_chats",
+  },
+  {
+    command: "chatid",
+    description: "Get the chat ID",
+    type: "all_private_chats",
+  },
+  {
+    command: "ban",
+    description: "Ban a user",
+    type: "all_group_chats",
+    isAdmin: true,
+  },
+  {
+    command: "setcommands",
+    description: "Set the bot's commands",
+    type: "chat",
+    isAdmin: true,
+  },
+  {
+    command: "setwebhook",
```

Comment:
Surprised cspell isn't complaining. Consider setWebhook etc

Reply:
I used camelCase for all of the commands but it's not allowing us and is throwing an error. I tried hyphenating originally and it caused the commands to fail. This seems to be enforced by telegram.

Docs again are surprisingly lacking in regards to their spec for things

https://core.telegram.org/constructor/botCommand
https://core.telegram.org/type/BotCommand

![image](https://github.com/user-attachments/assets/da842f35-27bc-45ca-9977-606bc751bfb9)

Reply:
`/setwebhook` or just `/webhook` it'll have to be